x-pack/plugins/enterprise_search/public/applications/app_search/components/result/result.tsx

```diff
@@ -49,23 +50,31 @@ export const Result: React.FC<Props> = ({
     if (schemaForTypeHighlights) return schemaForTypeHighlights[fieldName];
   };
 
+  const documentLink = getDocumentDetailRoute(resultMeta.engine, resultMeta.id);
   const conditionallyLinkedArticle = (children: React.ReactNode) => {
     return shouldLinkToDetailPage ? (
-      <ReactRouterHelper to={getDocumentDetailRoute(resultMeta.engine, resultMeta.id)}>
-        <a className="appSearchResult__content">{children}</a>
+      <ReactRouterHelper to={documentLink}>
+        <article className="appSearchResult__content appSearchResult__content--link">
+          {children}
+        </article>
       </ReactRouterHelper>
     ) : (
       <article className="appSearchResult__content">{children}</article>
     );
   };
 
+  const classes = classNames('appSearchResult', {
+    'appSearchResult--link': shouldLinkToDetailPage,
+  });
+
   return (
     <EuiPanel
       paddingSize="none"
-      className="appSearchResult"
+      className={classes}
       data-test-subj="AppSearchResult"
       title={i18n.translate('xpack.enterpriseSearch.appSearch.result.title', {
-        defaultMessage: 'View document details',
+        defaultMessage: 'Document {id}',
+        values: { id: result[ID].raw },
```

Comment:
I'm not 100% sold on this change and if it improves the experience for screenreader users at all - happy to revert if people aren't a fan of this